server/src/test/java/org/candlepin/dto/api/v1/SubscriptionTranslatorTest.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.candlepin.dto.AbstractTranslatorTest;
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.shim.ProductDataTranslator;
+import org.candlepin.model.Cdn;
+import org.candlepin.model.Certificate;
+import org.candlepin.model.Owner;
+import org.candlepin.model.SubscriptionsCertificate;
+import org.candlepin.model.dto.ProductData;
+import org.candlepin.model.dto.Subscription;
+import org.candlepin.util.Util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+
+import static org.apache.commons.collections.CollectionUtils.isEmpty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test suite for the ProductTranslator class
+ */
+public class SubscriptionTranslatorTest extends
+    AbstractTranslatorTest<Subscription, SubscriptionDTO, SubscriptionTranslator> {
+
+    private SubscriptionTranslator translator = new SubscriptionTranslator();
+
+    @Override
+    protected void initModelTranslator(ModelTranslator modelTranslator) {
+        modelTranslator.registerTranslator(new CdnTranslator(), Cdn.class, CdnDTO.class);
+        modelTranslator.registerTranslator(
+            new CertificateTranslator(), Certificate.class, CertificateDTO.class);
+        modelTranslator.registerTranslator(new NestedOwnerTranslator(), Owner.class, NestedOwnerDTO.class);
+        modelTranslator.registerTranslator(
+            new ProductDataTranslator(), ProductData.class, ProductDTO.class);
+        modelTranslator.registerTranslator(this.translator, Subscription.class, SubscriptionDTO.class);
+    }
+
+    @Override
+    protected SubscriptionTranslator initObjectTranslator() {
+        return this.translator;
+    }
+
+    @Override
+    protected Subscription initSourceObject() {
+        Subscription source = new Subscription();
+
+        source.setId("test_id");
+        source.setOwner(this.createOwner());
+        source.setProduct(this.createProduct());
+        source.setDerivedProduct(this.createProduct());
+        source.setProvidedProducts(this.createProducts());
+        source.setDerivedProvidedProducts(this.createProducts());
```

Comment:
There is no use for creating new util methods to create ProductData instances. You can create a `ProductDataTranslatorTest` instance and call `initSourceObject()` instead

Reply:
Done.